modules/ingest-common/src/internalClusterTest/java/org/elasticsearch/ingest/common/IngestRestartIT.java

```diff
@@ -326,4 +336,78 @@ public boolean validateClusterForming() {
         source = client().prepareGet("index", "fails").get(timeout).getSource();
         assertNull(source);
     }
+
+    /**
+     * This test is for confirming that forwarded bulk requests do not use system_write thread pool
+     * for non-system indexes. Before this fix, we were using system_write thread pool for all forwarded
+     * bulk requests causing the system_write thread pool to get overloaded.
+     */
+    public void testForwardBulkWithSystemWritePoolDisabled() throws Exception {
+        // Create a node with master only role and a node with ingest role
+        final String masterOnlyNode = internalCluster().startMasterOnlyNode();
+        final String ingestNode = internalCluster().startNode();  // What roles does a default node have ?
+
+        ensureStableCluster(2);
+
+        // Create Bulk Request
+        createIndex("index");
+
+        BytesReference source = new BytesArray("""
+            {
+              "processors" : [
+                  {"set" : {"field": "y", "value": 0}}
+              ]
+            }""");
+
+        PutPipelineRequest putPipelineRequest = new PutPipelineRequest("_id", source, XContentType.JSON);
+        clusterAdmin().putPipeline(putPipelineRequest).get();
+
+        int numRequests = scaledRandomIntBetween(32, 128);
+        BulkRequest bulkRequest = new BulkRequest();
+        BulkResponse response;
+        for (int i = 0; i < numRequests; i++) {
+            IndexRequest indexRequest = new IndexRequest("index").id(Integer.toString(i)).setPipeline("_id");
+            indexRequest.source(Requests.INDEX_CONTENT_TYPE, "x", 1);
+            bulkRequest.add(indexRequest);
+        }
+
+        // Block system_write thread pool on the ingest node
+        final ThreadPool ingestNodeThreadPool = internalCluster().getInstance(ThreadPool.class, ingestNode);
+        final var blockingLatch = new CountDownLatch(1);
+        // Can we use ThreadPoolStats instead of blocking the threadpool ?
```

Comment:
I personally prefer blocking system_write thread pool since it feels more direct to me. That said, checking `ThreadPoolStats` is a reasonable alternative and should make the test run faster. Happy for you to take the ThreadPoolStats approach. I am OK either way.

Reply:
I suggests system stats as another way to check. I don't really have a preference. Since the blocking mechanism has already been implemented here, I would say we just stick with that.

Reply:
And I would remove this comment in the PR update.

Reply:
Removed the comment. 
I will use blocking instead of stats for this one.